docs/source/common/lightning_cli.rst

```diff
@@ -0,0 +1,341 @@
+
+.. testcode::
+
+    from pytorch_lightning.utilities.cli import LightningCLI
+
+    cli = LightningCLI(MyModel, MyDataModule)
+
+The start of a possible implementation of :class:`MyModel` including the
+recommended argument descriptions in the docstring could be the one below. Note
+that by using type hints and docstrings there is no need to duplicate this
+information to define its configurable arguments.
+
+.. code-block:: python
+
+    class MyModel(LightningModule):
+
+        def __init__(
+            self,
+            encoder_layers: int = 12,
+            decoder_layers: List[int] = [2, 4]
+        ):
+            """Example encoder-decoder model
+
+            Args:
+                encoder_layers: Number of layers for the encoder
+                decoder_layers: Number of layers for each decoder block
+            """
+            ...
+
+With this model class, the help of the trainer tool would look as follows:
+
+.. code-block:: bash
+
+    $ python trainer.py --help
+    usage: trainer.py [-h] [--print_config] [--config CONFIG]
+                      [--trainer.logger LOGGER]
+                      ...
+
+    pytorch-lightning trainer command line tool
+
+    optional arguments:
+      -h, --help            show this help message and exit
+      --print_config        print configuration and exit
+      --config CONFIG       Path to a configuration file in json or yaml format.
+                            (default: null)
+
+    Customize every aspect of training via flags:
+      ...
+      --trainer.max_epochs MAX_EPOCHS
+                            Stop training once this number of epochs is reached.
+                            (type: int, default: 1000)
+      --trainer.min_epochs MIN_EPOCHS
+                            Force training for at least these many epochs (type: int,
+                            default: 1)
+      ...
+
+    Example encoder-decoder model:
+      --model.encoder_layers ENCODER_LAYERS
+                            Number of layers for the encoder (type: int, default: 12)
+      --model.decoder_layers DECODER_LAYERS
+                            Number of layers for each decoder block (type: List[int],
+                            default: [2, 4])
+
+The default configuration that option :code:`--print_config` gives is in yaml
+format and for the example above would look as follows:
+
+.. code-block:: bash
+
+    $ python trainer.py --print_config
+    model:
+      decoder_layers:
+      - 2
+      - 4
+      encoder_layers: 12
+    trainer:
+      accelerator: null
+      accumulate_grad_batches: 1
+      amp_backend: native
+      amp_level: O2
+      ...
+
+Note that there is a section for each class (model and trainer) including all
+the init parameters of the class. This grouping is also used in the formatting
+of the help shown previously.
+
+
+Trainer Callbacks and arguments with class type
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A very important argument of the
```

Comment:
we can keep 120 line length

Reply:
I believe he means "change all the rst file so that wrapping is at 120"